docs/examples/model_repository/resnet_torchscript/Readme.md

```diff
@@ -0,0 +1,154 @@
+
+```bash
+<model-repository-path>/
+  <model-name>/
+    config.pbtxt
+    1/
+      model.pt
+```
+
+
+```python
+model_dir = "sample_models_dir/resnet_torchscript/1" # making directory as required by Triton 
+if not os.path.exists(model_dir):
+    os.makedirs(model_dir)
+```
+
+First convert your model from GPU to CPU and then save it. This is recommended because the tracer may witness tensor creation on a specific device, so casting an already-loaded model may have unexpected effects. Casting the model before saving it ensures that the tracer has the correct device information.
+
+
+```python
+torch.jit.save(traced_model, os.path.join(model_dir,"model.pt"))
+```
+
+## Config File Creation
+
+Each model input and output must specify a name, datatype, and shape.
+
+The name specified for an input or output tensor must match the name expected by the model. PyTorch Naming Convention: Due to the absence of names for inputs and outputs in a TorchScript model, the `“name”` attribute of both the inputs and outputs in the configuration must follow a specific naming convention i.e. `“<name>__<index>”`. Where `<name>` can be any string and `<index>` refers to the position of the corresponding input/output. This means if there are two inputs and two outputs they must be named as: `“INPUT__0”`, `“INPUT__1”` and `“OUTPUT__0”`, `“OUTPUT__1”` such that `“INPUT__0”` refers to first input and `INPUT__1` refers to the second input, etc.
+
+Here is the config file that We will be using for the present Model
+
+``` json
+name: "resnet-torchscript"
+platform: "pytorch_libtorch"
+max_batch_size: 8
+input [
+  {
+    name: "input__0"
+    data_type: TYPE_FP32
+    format: FORMAT_NCHW
+    dims: [3,224,224 ]
+  }
+]
+output [
+  {
+    name: "output__0"
+    data_type: TYPE_FP32
+    dims: [ 1000 ]
+    label_filename: "resnet_labels.txt"
+  }
+]
+
+```
+
+`"resnet_labels.txt"` file has labels for the ImageNet data.
+
+## Loading Model to triton
+
+Load this model as `tritonserver --model-repository=sample_models_dir`
+
+## Doing Performance Test
+
+After you have Triton running you can send inference and other requests to it using the HTTP/REST or GRPC protocols from your Nvidia [Triton Client application](https://docs.nvidia.com/deeplearning/triton-inference-server/master-user-guide/docs/client_example.html).
+To simplify communication with Triton, the Triton project provides C++ and Python client libraries, and several example applications that show how to use these libraries.
+Nvidia Triton Client comes with the `perf_client` application is an essential tool for optimizing your model’s performance. Apart from below given two example, you can use many more performance tests as given at the [documentation page](https://docs.nvidia.com/deeplearning/triton-inference-server/master-user-guide/docs/optimization.html?highlight=perf_client).
```

Comment:
NVIDIA Triton Client SDK contains a performance analyzer (Point to perf_analyzer.md) which can be used to run experiments to measure inference throughput and latency in different scenarios.